pf9/modules/analytics_utils.py

```diff
@@ -0,0 +1,105 @@
+import analytics
+import uuid
+import math
+import time
+import datetime
+
+# the write_key is the Segment API authorization and identifier, without this no data submission will work. The API Key below is for Dev/Test work
+analytics.write_key = 't1tdSZocQ49Tx3G66lUjJXkOXjUSYRKx'
+
+def on_error(error, items):
+    print('An error occurred:', error)
+
+analytics.debug = True
+analytics.on_error = on_error
+
+
+class SegmentSessionWrapper:
+    def __init__(self, ctx):
+        self.ctx = ctx
+
+    def load_segment_session(self, segment_session, segment_event_properties, segment_subcommand):
+        """Load segment session related objects to ctx
+        """
+        self.ctx.params["segment_session"] = segment_session
+        self.ctx.params["segment_event_properties"] = segment_event_properties
+        self.ctx.params["segment_event_prefix"] = 'WZ PMKExpress CLI'
+        self.ctx.params["segment_subcommand"] = segment_subcommand
+        self.ctx.params["segment_step_count"] = 1
+
+    def send_track(self, step_name):
+        """ Builds event name based on subcommand and step count
+        """
+        event_name = " - ".join([self.ctx.params["segment_event_prefix"], self.ctx.params["segment_subcommand"],
+                              step_name, str(self.ctx.params["segment_step_count"])])
+        print("Dynamic event name: {}".format(event_name))
+        self.ctx.params["segment_step_count"] = self.ctx.params["segment_step_count"] + 1
+        self.ctx.params["segment_session"].send_track(event_name, self.ctx.params["segment_event_properties"], user_id=self.ctx.params['user_id'])
+
+    def send_track_error(self, step_name, error_message):
+        error_event_name = " - ".join([self.ctx.params["segment_event_prefix"], self.ctx.params["segment_subcommand"],
+                                 step_name, "ERROR"])
+        print("Error Dynamic event name: {}".format(error_event_name))
+        print("Error Details : {}".format(error_message))
+        segment_properties = self.ctx.params["segment_event_properties"].copy()
+        segment_properties.update(dict(error_message=error_message))
+        self.ctx.params["segment_session"].send_track(error_event_name, segment_properties,
+                                                      user_id=self.ctx.params['user_id'])
+
+
+class SegmentSession:
+
+    def __init__(self):
+        # unique for the device
+        self.device_id = str(uuid.uuid3(uuid.NAMESPACE_DNS, str(uuid.getnode())))
+
+        # unique for this session
+        self.anonymousId = str(uuid.uuid1())
+
+        # Session time is time in EPOC that must be passed to Segment in each track event to ensure that a single session is maintained. The client side CLI should generate this and submit the string with each API submission
+        self.session_time = math.floor(time.time())
```

Comment:
I don't understand the notion of using floor here. It is still a float.

Reply:
rounding off is apparently needed by amplitude. Fails otherwise.